Shared/Samples/Style features with custom dictionary/StyleFeaturesWithCustomDictionaryView.swift

```diff
@@ -0,0 +1,137 @@
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct StyleFeaturesWithCustomDictionaryView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The current custom dictionary style selection.
+    @State private var dictionaryStyleSelection: CustomDictionaryStyle = .file
+    
+    var body: some View {
+        VStack {
+            MapView(map: model.map)
+            
+            // The picker to toggle between style file and web style.
+            Picker("Dictionary Symbol Style", selection: $dictionaryStyleSelection) {
+                ForEach(CustomDictionaryStyle.allCases, id: \.self) { style in
+                    Text(style.label)
+                }
+            }
+            .pickerStyle(.segmented)
+            .padding()
+            .onChange(of: dictionaryStyleSelection) { _ in
+                // Update the feature layer with the correct dictionary renderer
+                // on selection change.
+                model.restaurantFeatureLayer.renderer = dictionaryStyleSelection == .web ? model.dictionaryRendererFromWebStyle : model.dictionaryRendererFromStyleFile
+            }
+        }
+    }
+}
+
+private extension StyleFeaturesWithCustomDictionaryView {
+    /// The view model for the sample.
+    class Model: ObservableObject {
+        /// A map with a topographic basemap centered on Esri in Redlands, CA.
+        let map: Map = {
+            let map = Map(basemapStyle: .arcGISTopographic)
+            map.initialViewpoint = Viewpoint(
+                latitude: 34.0543,
+                longitude: -117.1963,
+                scale: 1e4
+            )
+            return map
+        }()
+        
+        /// A feature layer with restaurants in Redlands, CA.
+        let restaurantFeatureLayer: FeatureLayer = {
+            let restaurantFeatureTable = ServiceFeatureTable(url: .redlandsRestaurants)
+            return FeatureLayer(featureTable: restaurantFeatureTable)
+        }()
+        
+        /// A dictionary renderer created from a custom symbol style dictionary file (.stylx) on local disk.
+        let dictionaryRendererFromStyleFile: DictionaryRenderer = {
+            // The URL to the symbol style dictionary file.
+            let restaurantStyleURL = URL.restaurants
+            
+            // Create the dictionary symbol style from the style file URL.
+            let restaurantStyle = DictionarySymbolStyle(url: restaurantStyleURL)
+            
+            // Create the dictionary renderer from the dictionary symbol style.
+            return DictionaryRenderer(dictionarySymbolStyle: restaurantStyle)
+        }()
+        
+        /// A dictionary renderer created from a custom symbol style hosted on ArcGIS Online.
+        let dictionaryRendererFromWebStyle: DictionaryRenderer = {
+            // The create a portal item with the id of the online web style.
+            let restaurantPortalItem = PortalItem(
```

Comment:
Repeating type information in the variable isn't that helpful:
```suggestion
            let restaurant = PortalItem(
```